src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -2494,19 +2494,15 @@ namespace OpenMS
       }
       else if (spec_view != nullptr)
       {
-        QTreeWidgetItem* item = spec_view->getTreeWidget()->currentItem();
-        if (item != nullptr)
+        ExperimentSharedPtrType new_exp_sptr(new ExperimentType());
+        if (spec_view->getSelectedScan(*new_exp_sptr))
```

Comment:
e.g., here copying the whole layer might be a more general solution

Reply:
jep, I agree. But currently we have no means to subselect from a full dataset. Or did I miss something?

Reply:
Do we need to? If so, I think this should then work differently (e.g., letting the user select spectra via check boxes (like curated in id view and export them). We have the file filter tool.

Reply:
I mean we can keep it but in my view a more general solution would be to copy the full layer and layer settings (aka current spectrum index)